app/services/npq/accept_application.rb

```diff
@@ -0,0 +1,112 @@
+
+    def call
+      return self unless valid?
+
+      ApplicationRecord.transaction do
+        teacher_profile.update!(trn: npq_application.teacher_reference_number) if npq_application.teacher_reference_number_verified?
+        create_participant_profile!
+        npq_application.update(lead_provider_approval_status: "accepted") && other_applications_in_same_cohort.update(lead_provider_approval_status: "rejected")
+        deduplicate_by_trn!
+      end
+
+      npq_application
+    end
+
+  private
+
+    def not_already_accepted
+      return if npq_application.blank?
+
+      errors.add(:npq_application, I18n.t("npq_application.has_already_been_accepted")) if npq_application.accepted?
+    end
+
+    def cannot_change_from_rejected
+      return if npq_application.blank?
+
+      errors.add(:npq_application, I18n.t("npq_application.cannot_change_from_rejected")) if npq_application.rejected?
+    end
+
+    def cohort
+      npq_application.cohort
+    end
+
+    def other_accepted_applications_with_same_course?
+      errors.add(:npq_application, I18n.t("npq_application.has_another_accepted_application")) if other_accepted_applications_with_same_course.present?
+    end
+
+    def other_accepted_applications_with_same_course
+      return if npq_application.blank?
+
+      @other_accepted_applications_with_same_course ||= NPQApplication
+                                                          .joins(:participant_identity)
+                                                          .where(participant_identity: { user_id: user.id })
+                                                          .where(npq_course: npq_course.rebranded_alternative_courses)
+                                                          .where(lead_provider_approval_status: "accepted")
+                                                          .where.not(id: npq_application.id)
+    end
+
+    def other_applications_in_same_cohort
+      @other_applications_in_same_cohort ||= NPQApplication
+        .joins(:participant_identity)
+        .where(participant_identity: { user_id: user.id })
+        .where(npq_course:)
+        .where(cohort:)
+        .where.not(id: npq_application.id)
+    end
+
+    def create_participant_profile!
+      ParticipantProfile::NPQ.create!(
+        id: npq_application.id,
+        schedule: NPQCourse.schedule_for(npq_course: npq_application.npq_course, cohort:),
+        npq_course: npq_application.npq_course,
+        teacher_profile:,
+        school_urn: npq_application.school_urn,
+        school_ukprn: npq_application.school_ukprn,
+        participant_identity: npq_application.participant_identity,
+      ).tap do |pp|
+        ParticipantProfileState.find_or_create_by(participant_profile: pp)
+      end
+    end
+
+    def participant_profile
+      @participant_profile ||= ParticipantProfile.find(npq_application.id)
+    end
+
+    def teacher_profile
+      @teacher_profile ||= user.teacher_profile || user.build_teacher_profile
```

Comment:
⚠️ ⚠️  Please change: Here you if the user does not have a teacher profile. 
You want to lookup teacher profile by by TRN and attach this `teacher_profile` to the user. In _theory_ there should not be two teacher profile with the same TRN. 
Another thing that needs to be though of is what if the user attach to the teacher profile is different than the one attached to the `NPQApplication` there should be some kind of error thrown as this has the potential to create duplicate profile!

@cwrw @peteryates @tonyheadford would you mind having a think about that, we've some some hacks in place to prevent double paying via the `TRN` routes but this looks like the place where duplicates could be unintentionally created?

At this point if it didn't exists you'd also want to add the TRN from `NPQApplication#teacher_reference_number` onto the `teacher_profile`

Reply:
chatted with @StupidCodeFactory in person but for posterity: We'll check this in detail and create a ticket incase to address as it's out of scope for this PR (I'll link the ticket after investigation in the comments below)